connector/connect/common/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -111,6 +111,9 @@ message SQL {
 
   // (Optional) A map of parameter names to literal expressions.
   map<string, Expression.Literal> args = 2;
+
+  // (Optional) A map of parameter names to column expressions.
+  map<string, Expression.UnresolvedAttribute> columns = 3;
```

Comment:
I am wondering if we can merge `map<string, Expression.Literal> args = 2;` and `map<string, Expression.UnresolvedAttribute> columns = 3;` to `map<string, Expression>`?

Reply:
BTW I am not sure if it can pass proto breaking changes check but re-using proto fields could be a good idea. Maybe originally the type of `args` should already use `Expression`.

Reply:
There are some different between args and columns. `SELECT {col} FROM t1 WHERE id=1` , the `{col}` used for `columns`. `SELECT id FROM t1 WHERE id= :value `, the `:value` used for `args`. The symbols used by the two are different, and the processing logic is also different. `{col}` is replaced before SQL analysis, and args is value replacement done in SQL analysis.

Reply:
It can also be merge together, but it doesn't look so clear on the interface.